cmd2.py

```diff
@@ -438,6 +438,229 @@ def strip_ansi(text):
+    2. 'sub cmd args' will call do_cmd(args) in a SubMenu instance
+    3. 'sub ... [TAB]' will have the same behavior as [TAB] in a SubMenu cmdloop
+       i.e., autocompletion works the way you think it should
+    4. 'help sub [cmd]' will print SubMenu's help (calls its do_help())
+    """
+
+    class _Nonexistent(object):
+        """
+        Used to mark missing attributes.
+        Disable __dict__ creation since this class does nothing
+        """
+        __slots__ = ()  #
+
+    def __init__(self,
+                 submenu,
+                 command,
+                 aliases=(),
+                 reformat_prompt="{super_prompt}>> {sub_prompt}",
+                 shared_attributes={},
+                 require_predefined_shares=True,
+                 create_subclass=False
+                 ):
+        """Set up the class decorator
+
+        submenu (Cmd):              Instance of something cmd.Cmd-like
+
+        command (str):              The command the user types to access the SubMenu instance
+
+        aliases (iterable):         More commands that will behave like "command"
+
+        reformat_prompt (str):      Format str or None to disable
+            if it's a string, it should contain one or more of:
+                {super_prompt}:     The current cmd's prompt
+                {command}:          The command in the current cmd with which it was called
+                {sub_prompt}:       The subordinate cmd's original prompt
+            the default is "{super_prompt}{command} {sub_prompt}"
+
+        shared_attributes (dict):   dict of the form {'subordinate_attr': 'parent_attr'}
+            the attributes are copied to the submenu at the last moment; the submenu's
+            attributes are backed up before this and restored afterward
+
+        require_predefined_shares: The shared attributes above must be independently
+            defined in the subordinate Cmd (default: True)
+
+        create_subclass: put the modifications in a subclass rather than modifying
+            the existing class (default: False)
+        """
+        self.submenu = submenu
+        self.command = command
+        self.aliases = aliases
+
+        if reformat_prompt is not None and not isinstance(reformat_prompt, str):
+            raise ValueError("reformat_prompt should be either a format string or None")
+        self.reformat_prompt = reformat_prompt
+
+        if require_predefined_shares:
+            for attr in shared_attributes.keys():
+                if not hasattr(submenu, attr):
+                    raise AttributeError("The shared attribute '{attr}' is not defined in {cmd}. Either define {attr} "
+                                         "in {cmd} or set require_predefined_shares=False."
+                                         .format(cmd=submenu.__class__.__name__, attr=attr))
+        self.shared_attributes = shared_attributes
+
+        self.create_subclass = create_subclass
+
+    def __call__(self, Cmd):
```

Comment:
Cmd arg name shadows cmd2.Cmd module name from outer scope, best to be clear and rename argument